models/models.js

```diff
@@ -19,14 +19,40 @@ exports.readArticles = (article_id) => {
     });
 };
 
-exports.readAllArticles = () => {
-  return db
-    .query(
-      "SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM ARTICLES LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC"
-    )
+exports.readAllArticles = (topic, sort_by = 'created_at', order="desc") => {
+  const acceptedTopics = ["mitch", "cats", "paper", undefined];
+  const queryValues = [];
+  const acceptedProperties = ['title', 'topic', 'author', 'body', 'created_at', 'votes'];
+  const acceptedOrders = ['asc', 'desc']
+
+  if (!acceptedTopics.includes(topic)) {
```

Comment:
You don't need to bad request if the topic is not found, it doesn't need to be a rigid as the others as this is not a column name and can be easily escaped using $1 syntax, so there is no risk of SQL injection therefore no need to have a list of acceptable topics.